node/src/main/scala/coop/rchain/node/proposescript/ProposeLeaderful.scala

```diff
@@ -0,0 +1,351 @@
+    proposeApi
+      .use(
+        proposeService =>
+          for {
+            proposeResResp <- proposeService.proposeResult(ProposeResultQuery())
+            _ <- Task
+                  .raiseError(
+                    new Exception(
+                      s"Getting propose result on $host:$grpcPort failed on ${proposeResResp.message.error}"
+                    )
+                  )
+                  .whenA(proposeResResp.message.isError || proposeResResp.message.result.isEmpty)
+            proposeRes = proposeResResp.message.result.get
+          } yield proposeRes
+      )
+  def isContainBlock(blockHash: String): Task[Boolean] =
+    getBlock(blockHash = blockHash).redeem(_ => false, _ => true)
+
+  private def deployAndPropose(isAsync: Boolean): Task[String] =
+    for {
+      latestBlocks  <- getLatestBlocks(depth = 1)
+      latestBlock   = latestBlocks.head
+      _             <- deploy(latestBlock.blockNumber)
+      proposeResult <- propose(isAsync = isAsync)
+    } yield proposeResult
+
+  def deployAndProposeAsync: Task[String] =
+    for {
+      result <- deployAndPropose(isAsync = true)
+      _ <- result match {
+            case ProposeAsyncSuccessRegex(seqNum) => seqNum.pure[Task]
+            case failedResult =>
+              Task.raiseError[String](
+                new Exception(s"Propose async failed with ${failedResult} on $host: $grpcPort")
+              )
+          }
+      proposeResult <- proposeResult
+      blockHash <- proposeResult match {
+                    case ProposeSuccessRegex(blockHash) => blockHash.pure[Task]
+                    case r =>
+                      Task.raiseError[String](
+                        new Exception(
+                          s"Getting async propose result failed with ${r} on $host: $grpcPort"
+                        )
+                      )
+                  }
+    } yield blockHash
+
+  def deployAndProposeSync: Task[String] =
+    for {
+      result <- deployAndPropose(isAsync = false)
+      blockHash <- result match {
+                    case ProposeSuccessRegex(blockHash) => blockHash.pure[Task]
+                    case proposeResult =>
+                      Task.raiseError[String](
+                        new Exception(
+                          s"Parsing propose result failed $proposeResult on $host:$grpcPort"
+                        )
+                      )
+                  }
+    } yield blockHash
+}
+
+final case class ProposeConfig(
+    leader: ValidatorInfo,
+    followers: List[ValidatorInfo],
+    checkInterval: FiniteDuration,
+    proposeInterval: FiniteDuration
+)
+
+final case class ProposeScriptOptions(arguments: Seq[String]) extends ScallopConf(arguments) {
+  val width = 120
+  helpWidth(width)
+  printedName = "propose-script"
+
+  val config = opt[Path](
+    descr = s"Target config file path",
+    required = true
+  )
+  verify()
+
+}
+
+object ProposeLeaderful {
+
+  def main(args: Array[String]): Unit = {
```

Comment:
Monix (as IO) has very useful helper for defining app entry point which can directly return Task.

This is what it might look like with `task` in F and repeated with Stream.
```scala
object ProposeLeaderful extends TaskApp {
  def run(args: List[String]): Task[ExitCode] = {
    ...
    Stream.eval(task).repeat.compile.drain.as(ExitCode.Success)
  }
```